_posts/2024-03-22-kraft-migration.md

```diff
@@ -0,0 +1,388 @@
+---
+layout: post
+title:  "From ZooKeeper to KRaft: let's automate the migration with Strimzi"
+date: 2024-03-22
+author: paolo_patierno
+---
+
+The Apache Kafka project has been using [Apache ZooKeeper](https://zookeeper.apache.org/) since its inception to store metadata.
+Registered brokers, the current controller and topics' configuration are just a few of the data that ZooKeeper stores for supporting a running Kafka cluster.
+Using a dedicated and centralized system for maintaining configuration data sounded the right approach at that time and ZooKeeper has been doing its work for long.
```

Comment:
```suggestion
Using a dedicated and centralized system for maintaining cluster metadata and offload leader election was the right approach at that time, because ZooKeeper was battle tested.
```